### DISCUSSION TITLE
How to extend the Keycloak ProtoSchema for custom cache types?

### DISCUSSION THREAD

**Discussion by simse07** [keycloak/keycloak#34912] [Q&A] [answered]
Hi everybody,
we have custom cache types defined and would like to update to Keycloak 26.
Due the change of the Infinispan marshalling, we are looking for a way to define marshallers for our custom types.

Does somebody has experience with this?
How can I register a marshaller at the SerializationContext?

Best regards

**Comment by pruivo** [comment, marked as answer]
After creating your `SerializationContextInitializer`, deploy the JAR in Keycloak and register the `SerializationContextInitializer` in the `cache-ispn.xml`. See https://infinispan.org/docs/stable/titles/encoding/encoding.html#registering-sci-embedded-caches_marshalling

**Reply by simse07** [reply]
Thank you very much!
